### ISSUE TITLE
Convert to ansible role

### ISSUE THREAD

**Issue by madoke** [hsanjuan/ansible-ipfs-cluster#9]
Hi ! This is a great playbook and i've been using it for a while now, however i feel that it could be improved by converting it to the normal role format so that it could be imported on other playbooks and used alongside other roles. Will be glad to contribute if this makes sense to the project

**Comment by hsanjuan** [comment]
Makes sense in principle. Can you outline what changes that will require?

**Comment by madoke** [comment]
> Makes sense in principle. Can you outline what changes that will require?

ok, so accorfing to the docs (https://galaxy.ansible.com/docs/contributing/creating_role.html),  we need to change the directory structure into this:
```
|-- README.md
|-- defaults
|   `-- main.yml
|-- files
|-- handlers
|   `-- main.yml
|-- meta
|   `-- main.yml
|-- tasks
|   `-- main.yml
|-- templates
|-- tests
|   |-- inventory
|   `-- test.yml
`-- vars
    `-- main.yml
```

Which would mean:
- move all the tasks/handlers/templates under roles to the new structure (maybe ipfs + ipfs-cluster folders inside each ?)
- move all the vars under host_vars / group_vars to role parameters/defaults
- adapt the readme file with instructions on how to use the role
- implement some tests (optional)

Maybe i can create a draft pull request and we can go from there ?

**Comment by madoke** [comment]
PR #10 complete. Can you give it a try ?

**Comment by insanity54** [comment]
Any progress on this? Anything I can help with?

**Comment by madoke** [comment]
> Any progress on this? Anything I can help with?

Hey, were you able to use it ? I've been using this branch occasionally and so far so good, but i'd say it could probably use some updates now. Eitherway, let me know if it works for you and if its worth to get these updates done so we can merge

**Comment by insanity54** [comment]
Yes I was, thank you! Here's what I'm using in my requirements.yml

```
---

roles:
  - name: madoke.ansible-ipfs-cluster
    src: https://github.com/madoke/ansible-ipfs-cluster.git
    scm: git
    type: git
    version: 3bd23433fdae86f342eeb23a5857119a87adbac9
```

It's working quite well and I appreciate the time savings it has afforded me. Right now I'm pulling in madoke.ansible-ipfs-cluster via git, but It would be wonderful to have ansible-ipfs-cluster on ansible-galaxy. If there's anything that needs doing to make it happen, I am happy to contribute.

**Comment by mbwmbw1337** [comment]
Any updates on this? Would love to contribute!

**Comment by madoke** [comment]
> Any updates on this? Would love to contribute!

It's probably outdated now and some parts may need revisiting, but it did the job. Let me know if you have trouble running it I can try to help !
